lib/flip_card.dart

```diff
@@ -175,6 +173,8 @@ class FlipCardState extends State<FlipCard>
 
   @override
   Widget build(BuildContext context) {
+    widget.controller?.state = this;
```

Comment:
Assigning variables in `build()` is a code smell, because it can potentially be triggered 60+ times per second and also alters the state (which makes subsequent build calls behave potentially different).

If this was assigned via the `??=` operator then in this specific case I don't see any practical problems.

Reply:
thanks for the good point

Reply:
thanks for the good point